x-pack/legacy/plugins/code/public/components/commits/commit.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import {
+  EuiButtonIcon,
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiPanel,
+  EuiText,
+  EuiTextColor,
+  EuiCopy,
+  EuiTitle,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { CommitInfo } from '../../../model/commit';
+import { MAIN_ROOT } from '../routes';
+import { PathTypes } from '../../common/types';
+
+const COMMIT_ID_LENGTH = 8;
+
+interface ActionProps {
+  repoUri: string;
+  commitId: string;
+}
+
+const copyLabel = i18n.translate('xpack.code.commits.copyCommitSha', {
+  defaultMessage: 'Copy the full commit ID',
+});
+const revisionLinkLabel = i18n.translate('xpack.code.commits.revisionLink', {
+  defaultMessage: 'View the project at this commit',
+});
+
+const CommitActions = ({ commitId, repoUri }: ActionProps) => {
+  const revisionUri =
```

Comment:
Here's the first question: Is this the best way to generate a link to this page? I saw a lot of different methods for URI generation; this was the first one that worked.

Reply:
@WangQianliang please advise, if we have multiple way to do common link like this, should we provide common method?

Reply:
@WangQianliang bump

Reply:
It would be best if we provide some methods to do this kind of things. Or here I think 
```
`#${repoUri}/${PathTypes.tree}/${commitId}`
```
would be fine.

Reply:
👍 thanks for the feedback!